examples/test_steps_blending_rmi.py

```diff
@@ -0,0 +1,287 @@
+
+from matplotlib import cm, pyplot as plt
+import numpy as np
+import xarray as xr
+import os
+from datetime import datetime
+from pprint import pprint
+from pysteps import io, rcparams
+from pysteps.utils import conversion, transformation, reprojection
+from pysteps.visualization import plot_precip_field
+
+# Set workdir and number of output time steps
+work_dir = "c:\\Users\\imhof_rn\\pysteps"
+os.chdir(work_dir)
+n_timesteps = 36
+
+# Give date as input
+date_str = "202107141010"  # input("Give date and time (e.g.: 201609281600):    ")
+
+date = datetime.strptime(date_str, "%Y%m%d%H%M")
+data_source = rcparams.data_sources["rmi"]
+
+# Import the radar composite to reproject NWP
+root_path = data_source["root_path"]
+path_fmt = data_source["path_fmt"]
+fn_pattern = data_source["fn_pattern"]
+fn_ext = data_source["fn_ext"]
+timestep = data_source["timestep"]
+importer_name = data_source["importer"]
+importer_kwargs = data_source["importer_kwargs"]
+
+# Find the radar files in the archive
+fns = io.find_by_date(
+    date, root_path, path_fmt, fn_pattern, fn_ext, timestep, num_prev_files=2
+)
+
+importer = io.get_method(importer_name, "importer")
+R_input = io.read_timeseries(fns, importer, legacy=False, **importer_kwargs)
+
+radar_data_xr = R_input[-1, :, :]
+radar_metadata = radar_data_xr.x.attrs.copy()
+radar_metadata.update(**radar_data_xr.y.attrs)
+radar_metadata.update(**radar_data_xr.attrs)
+
+# PLOTTING
+
+# R, _, metadata = io.import_odim_hdf5(filename, "RATE", legacy=True, **importer_kwargs)
+# Convert to rain rate
+# R, metadata = conversion.to_rainrate(R, metadata)
+# Nicely print the metadata
+# pprint(metadata)
+# Plot the rainfall field
+# plot_precip_field(R, geodata=metadata)
+# plt.show()
+
+
+# Import the NWP data
+date_str = "202107140600"  # input("Give date and time (e.g.: 201609281600):    ")
+
+date = datetime.strptime(date_str, "%Y%m%d%H%M")
+nwp_data_source = rcparams.data_sources["rmi_nwp"]
+filename = os.path.join(
+    nwp_data_source["root_path"],
+    datetime.strftime(date, nwp_data_source["path_fmt"]),
+    datetime.strftime(date, nwp_data_source["fn_pattern"])
+    + "."
+    + nwp_data_source["fn_ext"],
+)
+
+nwp_data_xr = io.import_rmi_nwp_xr(filename)
+nwp_metadata = nwp_data_xr.x.attrs.copy()
+nwp_metadata.update(**nwp_data_xr.y.attrs)
+nwp_metadata.update(**nwp_data_xr.attrs)
+
+# plot_precip_field(nwp_data_xr.values[1,:,:], geodata=nwp_metadata)
+# plt.show()
+
+# Reproject the projection of the nwp data to the projection of the radar data
+R_NWP = reprojection(nwp_data_xr[0 : n_timesteps + 1, :, :], radar_data_xr)
+nwp_rep_metadata = R_NWP.x.attrs.copy()
+nwp_rep_metadata.update(**R_NWP.y.attrs)
+nwp_rep_metadata.update(**R_NWP.attrs)
+
+# plot_precip_field(R_NWP[0,:,:], geodata=nwp_rep_metadata)
+# plt.show()
+
+########## Now start the blending! ################
+
+import pysteps
+from pysteps import cascade, blending
+
+R_input.data[R_input.data < radar_metadata["threshold"]] = 0.0
```

Comment:
usual annoying comment: let's try to follow python's naming convention more strictly than in the past: var names use snake_case